ci/travis_before_script_c_glib.sh

```diff
@@ -65,6 +65,9 @@ popd
 
 pushd $ARROW_C_GLIB_DIR
 
+export ARROW_C_GLIB_CFLAGS="-DARROW_NO_DEPRECATED_API"
+export ARROW_C_GLIB_CXXFLAGS="-DARROW_NO_DEPRECATED_API"
```

Comment:
How about calling `./configure` with `CFLAGS` and `CXXFLAGS` like the following?:

```shell
CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS CFLAGS=-DARROW_NO_DEPRECATED_API"
CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS CXXFLAGS=-DARROW_NO_DEPRECATED_API"
./configure $CONFIGURE_OPTIONS
```

If we use generic `configure` options, we don't need to introduce Arrow specific variable such as `ARROW_C_GLIB_CFLAGS`.